src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js

```diff
@@ -278,64 +278,58 @@ PrimeFaces.widget.DatePicker = PrimeFaces.widget.BaseWidget.extend({
         var _self = this;
         this.cfg.onViewDateChange = function(event, date) {
             _self.viewDateOption = date;
-
-            if(_self.hasBehavior('viewChange')) {
-                _self.fireViewChangeEvent(date.getFullYear(), date.getMonth());
-            }
-            if (_self.cfg.lazyModel) {
-                var options = {
-                    source: _self.id,
-                    process: _self.id,
-                    update: _self.id,
-                    formId: _self.cfg.formId,
-                    params: [
-                        {name: _self.id + '_year', value: date.getFullYear()},
-                        {name: _self.id + '_month', value: date.getMonth()}
-                    ],
-                    onsuccess: function(responseXML, status, xhr) {
-                        PrimeFaces.ajax.Response.handle(responseXML, status, xhr, {
-                            widget: _self,
-                            handle: function(content) {
-                                var dateMetadata = JSON.parse(content).dateMetadata;
-                                var disabledDates = [];
-                                for (date in dateMetadata) {
-                                    if (dateMetadata[date].disabled) {
-                                        disabledDates.push(date);
-                                    }
-                                }
-                                _self.setDisabledDates(disabledDates);
-                            }
-                        });
-
-                        return true;
-                    }
-                };
-                PrimeFaces.ajax.Request.handle(options);
-            }
+            _self.fireViewChangeEvent(date);
         };
     },
 
     /**
      * Triggers the event for when the date picker changed to a different month or year page.
      * @private
-     * @param {number} year The year to which the date picker changed.
-     * @param {number} month The year to which the date picker changed, starting with `0` for `Janurary`.
+     * @param {Date} date The date to which the date picker changed.
      */
-    fireViewChangeEvent: function(year, month) {
-        if(this.cfg.behaviors) {
+    fireViewChangeEvent: function(date) {
+        var _self = this, fired = false;
+        var options = {
+            params: [
+                {name: this.id + '_year', value: date.getFullYear()},
+                {name: this.id + '_month', value: date.getMonth()}
+            ]
```

Comment:
I think it might be becaue you are missing:
```javascript
source: _self.id,
                    process: _self.id,
                    update: _self.id,
                    formId: _self.cfg.formId,
```

Reply:
Yes, update did the job!